oa-enterprise/lib/omniauth/strategies/cas.rb

```diff
@@ -27,21 +48,36 @@ def request_phase
       end
 
       def callback_phase
-        ticket = request.params['ticket']
-        return fail!(:no_ticket, 'No CAS Ticket') unless ticket
-        validator = ServiceTicketValidator.new(@configuration, callback_url, ticket)
-        @user_info = validator.user_info
-        return fail!(:invalid_ticket, 'Invalid CAS Ticket') if @user_info.nil? || @user_info.empty?
+        @ticket = request.params['ticket']
+        return fail!(:no_ticket, 'No CAS Ticket') unless @ticket
+
+        self.raw_info = ServiceTicketValidator.new(@configuration, callback_url, @ticket).user_info
+
+        return fail!(:invalid_ticket, 'Invalid CAS Ticket') if raw_info.empty?
+
         super
       end
 
       def auth_hash
         OmniAuth::Utils.deep_merge(super, {
-          'uid' => @user_info.delete('user'),
-          'extra' => @user_info
+          'uid' => @raw_info['user'],
```

Comment:
configure omniauth object as defined in [`Auth Hash Schema`](https://github.com/omniauth/omniauth/wiki/Auth-Hash-Schema)